config/config/src/main/java/io/helidon/config/ConfigHelper.java

```diff
@@ -43,7 +43,7 @@ private ConfigHelper() {
                 .filter(e -> e.getValue() instanceof ValueNodeImpl)
                 .collect(Collectors.toMap(
                         e -> e.getKey().toString(),
-                        e -> Config.Key.escapeName(((ValueNodeImpl) e.getValue()).get())
+                        e -> ((ValueNodeImpl) e.getValue()).get()
```

Comment:
Why is this change done?

Reply:
There was this issue with escaped dot in config value:
```java
Map<String, String> values = Map.of("key.some", "jdbc:h2:file:./target/lra-coordinator");
        org.eclipse.microprofile.config.spi.ConfigSource mpSource = MpConfigSources.create(ConfigSources.create(values).build());
        assertThat(mpSource.getValue("key.some"), is("jdbc:h2:file:./target/lra-coordinator"));
```
```
Expected :jdbc:h2:file:./target/lra-coordinator
Actual   :jdbc:h2:file:~1/target/lra-coordinator
```
Escape is moved here:
https://github.com/oracle/helidon/pull/3769/files#diff-9d50a6dc46f6009326acc97b03ec1f183f0018408926cfb98eb236ebb8c85951R184